package com.tattvam.registration;

import java.io.IOException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;

import javax.security.auth.message.callback.PrivateKeyCallback.Request;
import javax.servlet.RequestDispatcher;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;

/**
 * Servlet implementation class Login
 */
@WebServlet("/login")
public class Login extends HttpServlet {
	private static final long serialVersionUID = 1L;
    private static final String JDBC_URL = "jdbc:mysql://127.0.0.1:3306/webapp?useSSL=false";
    private static final String USERNAME = "root";
    private static final String PASSWORD = "pass123";

	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		String username = request.getParameter("username");
		String password = request.getParameter("password");
		Connection con = null;
		PreparedStatement pst = null;
		HttpSession session=request.getSession();
		RequestDispatcher dispatcher= null;
		
		try {
			Class.forName("com.mysql.cj.jdbc.Driver");

            con = DriverManager.getConnection(JDBC_URL, USERNAME, PASSWORD);
            pst = con.prepareStatement("SELECT * FROM users WHERE username = ? AND password = ?");
            pst.setString(1, username);
            pst.setString(2, password);
            
            ResultSet rs = pst.executeQuery();
            if (rs.next()) {
               session.setAttribute("username", rs.getString("username"));
               dispatcher= request.getRequestDispatcher("login.jsp");
            } else {
                request.setAttribute("status", "failed");
                dispatcher= request.getRequestDispatcher("login.jsp");
            }
            dispatcher.forward(request, response);
			
		} catch (Exception e) {
			e.printStackTrace();
		}

	}

}
